wrapanapi/systems/msazure.py

```diff
@@ -814,6 +839,25 @@ def remove_discs_by_search(self, disc_name):
             self.logger.debug('No discs matching {} were found'.format(disc_name))
         return results
 
+    def remove_disc_by_search(self, disc_name, resource_group=None):
```

Comment:
This looks like it's re-implementing the method defined at L825, just with singular vs plural naming?

Reply:
It's little different, it accepts resource group parameter and calculates data by using resource group.

Reply:
Probably I can rename the def name to `remove_disc_by_resource_group`, basically I dont want to modify existing def used by many !

Reply:
@mshriver Ok as per discussion on GChat, I am keeping the flavor of existing def `remove_discs_by_search` with the extra spice/parameter of resource_group. 

I have kept the existing method of finding the unused disc as well as my newly added , let me know if you want to stick to my newest way of doing it.